src/DotNetWorker/Hosting/FunctionsApplicationBuilder.cs

```diff
@@ -0,0 +1,134 @@
+    public ConfigurationManager Configuration => _hostApplicationBuilder.Configuration;
+
+    /// <inheritdoc/>
+    public IHostEnvironment Environment { get; private set; }
+
+    /// <inheritdoc/>
+    public ILoggingBuilder Logging => _hostApplicationBuilder.Logging;
+
+    /// <inheritdoc/>
+    public IMetricsBuilder Metrics => _hostApplicationBuilder.Metrics;
+
+    /// <inheritdoc />
+    public IServiceCollection Services => _hostApplicationBuilder.Services;
+
+    /// <summary>
+    /// Only available to AspNetCore extension (and maybe not even there in the future)
+    /// </summary>
+    internal IHostBuilder HostBuilder => _bootstrapHostBuilder;
+
+    /// <summary>
+    /// Run the given actions to initialize the host. This can only be called once.
+    /// </summary>
+    /// <returns>An initialized <see cref="IHost"/>.</returns>
+    public IHost Build()
+    {
+        if (_hasBuilt)
+        {
+            throw new InvalidOperationException("The application has already been built.");
+        }
+
+        // we skipped this; make sure we add it at the end
+        this.UseDefaultWorkerMiddleware();
+
+        _hasBuilt = true;
+
+        return _hostApplicationBuilder.Build();
+    }
+
+    /// <inheritdoc/>
+    public void ConfigureContainer<TContainerBuilder>(IServiceProviderFactory<TContainerBuilder> factory, Action<TContainerBuilder>? configure = null) where TContainerBuilder : notnull
+    {
+        _hostApplicationBuilder.ConfigureContainer(factory, configure);
+    }
+
+    private IFunctionsWorkerApplicationBuilder InitializeHosting(BootstrapHostBuilder bootstrapHostBuilder)
+    {
+        bootstrapHostBuilder.ApplyServiceProviderFactory();
+        bootstrapHostBuilder.RunDefaultCallbacks();
+
+        IFunctionsWorkerApplicationBuilder functionsWorkerApplicationBuilder = null!;
+
+        foreach (var descriptor in Services)
+        {
+            if (descriptor.ServiceType == typeof(IFunctionsWorkerApplicationBuilder))
+            {
+                functionsWorkerApplicationBuilder = (IFunctionsWorkerApplicationBuilder)descriptor.ImplementationInstance!;
+                break;
+            }
+        }
+
+        // Grab the HostBuilderContext from the property bag to use in the ConfigureWebHostBuilder. Then
+        // grab the IWebHostEnvironment from the webHostContext. This also matches the instance in the IServiceCollection.
+        //var hostContext = (HostBuilderContext)bootstrapHostBuilder.Properties[typeof(HostBuilderContext)];
+        Environment = bootstrapHostBuilder.Context.HostingEnvironment;
+
+        //Host = new ConfigureHostBuilder(bootstrapHostBuilder.Context, Configuration, Services);
+        //WebHost = new ConfigureWebHostBuilder(webHostContext, Configuration, Services);
+
+        return functionsWorkerApplicationBuilder;
+    }
+
+    /// <inheritdoc/>
+    public IFunctionsWorkerApplicationBuilder Use(Func<FunctionExecutionDelegate, FunctionExecutionDelegate> middleware)
```

Comment:
Is an equivalent needed for the other variations like `UseWhen`, etc.? Or will those call through here somehow?

Reply:
`UseWhen()` is an extension method off of `IFunctionsWorkerApplicationBuilder`, so it works with this too.

Reply:
right, and it does indeed seem like it calls through here, so we still get the check for if it's built.  👍